### DISCUSSION TITLE
How to check if YamlScalar is a string

### DISCUSSION THREAD

**Discussion by wakaztahir** [charleskorn/kaml#412] [Q&A] [answered]
The JsonPrimitive has isString property which is a boolean , but YamlScalar doesn't

How would I know if YamlScalar is a string

**Comment by charleskorn** [comment, marked as answer]
YAML does not have an equivalent concept - in YAML, `"3"` is a valid integer, for example.

Could you perhaps expand on what you're trying to do?

**Reply by wakaztahir** [reply]
I was trying to convert yaml to json tree 

I've created a basic yaml converter as a workaround for a problem I cannot solve yet

It's working , It tries booleans then numbers and lastly sting conversion

It is available in
 https://github.com/Qawaz/K-OpenApi-Parser/tree/main/yaml-parser

**Reply by charleskorn** [reply]
> It's working , It tries booleans then numbers and lastly sting conversion

That's going to be the easiest way I think. 

Keep in mind that kaml is not intended to be a general-purpose YAML parsing library. If you're looking for that, SnakeYAML is a better fit.
